src/core/input_descriptor.rs

```diff
@@ -416,12 +417,35 @@ pub struct ConstraintsField {
     predicate: Option<Predicate>,
     #[serde(skip_serializing_if = "Option::is_none")]
     filter: Option<serde_json::Value>,
+    // NOTE: Skip the JSON schema from serialization,
+    // and use the filter value for constructing the schema.
+    // NOTE: This will mean the validation will be None on
+    // deserialization, and will need to be set manually.
+    #[serde(skip)]
+    // NOTE: Wrapping JSONSchema in an Arc for cloning.
+    validator: Option<Arc<JSONSchema>>,
```

Comment:
This is going to make it very difficult to construct a presentation definition on the verifier side. Instead, can you replace this with a new type?
```suggestion
    validator: ConstraintsFieldValidator
```

```rust
pub struct ConstraintsFieldValidator {
    raw: serde_json::Value,
    compiled: JSONSchema,
}

impl Serialize for ConstraintsFieldValidator {
    // Just serialize `self.raw`
}

impl Deserialize for ConstraintsFieldValidator {
    // Do `serde_json::Value::deserialize(d)` and then try to construct the JSONSchema.
}
```

Reply:
I see your comment was that you were concerned about time. I'll do up a commit for this change.